src/Panel/Ui/Buttons/Button.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Kirby\Panel\Ui\Buttons;
+
+use Closure;
+use Kirby\Cms\App;
+use Kirby\Panel\Panel;
+use Kirby\Toolkit\Controller;
+
+/**
+ * A single view button
+ * @since 5.0.0
+ *
+ * @package   Kirby Panel
+ * @author    Nico Hoffmann <nico@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ */
+readonly class Button
+{
+	public function __construct(
+		protected string $view,
+		protected string|array|Closure $config
+	) {
+	}
+
+	/**
+	 * Resolves a button defined as closure
+	 */
+	public function call(
+		Closure $button,
+		array $args = []
+	): array|null {
+		$kirby      = App::instance();
+		$controller = new Controller($button);
+
+		return $controller->call(data: [
+			'kirby' => $kirby,
+			'site'  => $kirby->site(),
+			'user'  => $kirby->user(),
```

Comment:
`Question`: what do you think about these default parameters/arguments to always pass to the Closure?

Reply:
If this usage risky for the performance, what about using `LazyValue`?

Reply:
@afbora I checked that but we also only use it as wrapper in `App::collection` for `pages` and `users`, not for these more basic objects.